### DISCUSSION TITLE
max_time follows an anti-pattern in SLIM

### DISCUSSION THREAD

**Discussion by remiadon** [scikit-mine/scikit-mine#138] [Q&A]
A max_time argument has been added to SLIM in 1bf3e3128b5820f16c76a40b144972f74ae7c412

As I explained in https://github.com/scikit-mine/scikit-mine/issues/121 
it's an anti-pattern from the point of view of reproducible research.

When porting research methods to meet industry-level standards, we want algorithms providing the exact same output for a given input, **regardless of the execution environment** 

If I take machine 1 and machine 2, with different RAM/CPU, can we guarantee that fitting a SLIM miner with eg. max_time=30 will give access to the exact code table ?

**Comment by thomasbtnfr** [comment]
Hello @remiadon,
I had well taken into account the issue #121. Indeed, the codetables may not be the same in this case. However, SLIM is an algorithm without parameters (except for pruning in theory) and the initial **k** parameter was not clear as mentioned in issue #121 that's why after internal discussion it was chosen to add a **max_time** attribute. 

I agree that this may not be ideal as it is not reproducible but this setting has the advantage of being clear to the user, perhaps a compromise could be found (to be discuss with @alexandre-termier and @PeggyCellier).
